### PR TITLE
Fixed SQS remove-statements

### PR DESCRIPTION
**Pull request by hav3wb**
Remove-statements action was failing when removing the last permission from an SQS. Updated to use boto's "remove_permission" instead of "set_queue_attribute". Tested successfully in sandbox. Addresses issue #1816.

### CODE REVIEW

**Comment by kumarsiyer** [comment]
hey @hav3wb any luck with SNS remove-statements ? #1779

**Comment by hav3wb** [comment]
@kumarsiyer I tried recreating the error you received for SNS and didn't receive that error -- everything appeared to be working. Therefore, I didn't include it in this PR as didn't know if it was a widespread issue. I did however check if this SQS fix would work for SNS (even though I wasn't seeing the error) and it appeared to work there as well. If this SQS fix ends up being approved, then I can open up a PR for SNS since I already have the fix for that as well (even though for me, functionality did not change). Do you happen to have an instance of why SNS was failing? (i.e. Removing the last permission or trying to remove multiple permissions at once?) So I can recreate the issue to test against.

**Comment by kumarsiyer** [comment]
@hav3wb thank you, let me give you more details as requested.

File: c7n/resources/sqs.py [hunk 1]
```diff
@@ -157,12 +157,17 @@ def process_resource(self, client, resource):
         if not found:
             return
 
-        client.set_queue_attributes(
-            QueueUrl=resource['QueueUrl'],
-            Attributes={
-                'Policy':json.dumps(p)
-            }
-        )
+        # client.set_queue_attributes(
```
Comment (jhnlsn) [inline c7n/resources/sqs.py]: @hav3wb can you remove the dead code

File: c7n/resources/sqs.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ class RemovePolicyStatement(RemovePolicyBase):
                     statement_ids: matched
     """
 
-    permissions = ('sqs:GetQueueAttributes', 'sqs:SetQueueAttributes')
+    permissions = ('sqs:GetQueueAttributes', 'sqs:SetQueueAttributes', 'sqs:RemovePermission')
```
Comment (jhnlsn) [inline c7n/resources/sqs.py]: please remove the invalid permissions as well

**Comment by hav3wb** [comment]
@jhnlsn Removed both

**Comment by kapilt** [comment]
this could use a unit test that tests the error reported.

**Comment by hav3wb** [comment]
@kapilt I didn't change any of the error reporting. The line I changed is still tested (and technically increased coverage as is 2 lines not 1) and the errors are already marked as covered according to coverage/coveralls.

**Comment by jhnlsn** [comment]
@hav3wb we do have tests where we remove 1 of 2 matched attributes, however, a test does not exist where we remove the permission when only 1 exists.  Would you be willing to write that up?

**Comment by hav3wb** [comment]
@jhnlsn I added a test

**Comment by kapilt** [comment]
@hav3wb fwiw, its much nicer generally to use feature branches on pull requests, your commit history on this pull request, has lots of extraneous commits unrelated to the pull request.

**Comment by hav3wb** [comment]
@kapilt Thanks! I actually realized that midway through this process and already set to fixing my branches for future updates

**Pull request merged**